src/org/zaproxy/zap/extension/custompages/CustomPage.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.custompages;
+
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.utils.EnableableInterface;
+
+public interface CustomPage extends EnableableInterface{
+
+	/**
+	 * Returns the context ID for a Custom Page.
+	 * 
+	 *  @return the context ID {@code int}
+	 */
+	int getContextId();
+
+	/**
+	 * Sets the context ID for a Custom Page.
+	 * 
+	 * @param contextId the ID of the context
+	 */
+	void setContextId(int contextId);
+
+	/**
+	 * Returns the content for a Custom Page.
+	 * 
+	 * @return the content {@code String}
+	 */
+	String getContent();

```

Comment:
I think the terms `Content` and  `ContentType` are a bit confusing.
These terms are strongly related to the http wording. 

How about renaming them into `Matcher` and `MatcherType` or  `PageMatcher` and `PageMatcherType` ? This renaming is also related to the class `CustomPageContentType`

Reply:
I've left the UI as Content and Content Type for the time being, however, I've changed all the code and javadoc.